src/frontend/src/optimizer/heuristic_optimizer.rs

```diff
@@ -81,6 +88,32 @@ impl<'a> HeuristicOptimizer<'a> {
     pub fn get_stats(&self) -> &Stats {
         &self.stats
     }
+
+    #[cfg(debug_assertions)]
+    fn check_equivalent_plan(rule: &BoxedRule, input_plan: &PlanRef, output_plan: &PlanRef) {
+        let fail = || {
+            panic!("{} fails to generate equivalent plan.\nInput schema: {:?}\nInput plan: \n{}\nOutput schema: {:?}\nOutput plan: \n{}\nSQL: {}",
+                   rule.description(),
+                   input_plan.schema(),
+                   input_plan.explain_to_string().unwrap(),
+                   output_plan.schema(),
+                   output_plan.explain_to_string().unwrap(),
+                   output_plan.ctx().sql());
+        };
+        if input_plan.schema().len() != output_plan.schema().len() {
+            fail();
+        }
+        for (a, b) in input_plan
+            .schema()
+            .fields
+            .iter()
+            .zip_eq_fast(output_plan.schema().fields.iter())
+        {
+            if a.data_type != b.data_type {
+                fail();
+            }
+        }
```

Comment:
Can we just derive `Eq` for `Schema`?

Reply:
```rust
#[derive(Clone, PartialEq, Eq, Hash)]
pub struct Field {
    pub data_type: DataType,
    pub name: String,
    /// For STRUCT type.
    pub sub_fields: Vec<Field>,
    /// The user-defined type's name, when the type is created from a protobuf schema file,
    /// this field will store the message name.
    pub type_name: String,
}
```

Reply:
I don't rhonk so, name is part of schema